frontend/src/components/Dashboard.tsx

```diff
@@ -1,35 +1,139 @@
 import styles from "../css/Dashboard.module.css";
 import { useUser } from "../contexts/UserContext";
 import { useNavigate } from "react-router-dom";
-import { logout } from "../utils";
+import {
+    acceptFriendRequest,
+    declineFriendRequest,
+    getAllData,
+    getIncomingFriendRequests,
+    logout,
+} from "../utils";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import {
     faUser,
     faChevronDown,
     faArrowRightLong,
 } from "@fortawesome/free-solid-svg-icons";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import LoggedOut from "./LoggedOut";
 import { APP_TITLE } from "../constants";
+import type { AllUserData, FriendRequestWithProfile } from "../types";
+import Modal from "./Modal";
 
-// Landing page; allows navigating to profile, map, etc.
+/**
+ *
+ * @returns A landing page where the user can view notifications and navigate out to other pages
+ */
 const Dashboard = () => {
+    // the logged-in user
     const { user } = useUser();
 
+    // logout the user and navigate to login page
+    const handleLogout = () => {
+        logout();
+        navigate("/login");
+    };
+
     const navigate = useNavigate();
 
+    // whether to show profile menu
     const [showingMenu, setShowingMenu] = useState(false);
 
-    const handleLogout = () => {
-        logout();
-        navigate("/login");
+    // user data to show in modal; null when modal is hidden
+    const [modalData, setModalData] = useState<AllUserData | null>(null);
+
+    // all active incoming friend requests
+    const [friendRequests, setFriendRequests] = useState(
+        Array() as FriendRequestWithProfile[],
+    );
+
+    // load active friend requests with data on the user who sent them
+    const loadFriendRequests = async () => {
+        const requests = await getIncomingFriendRequests();
+
+        const result = Array() as FriendRequestWithProfile[];
+
+        // iterate over each request, adding an object with its data and profile data to result
+        for (const request of requests) {
+            const data = await getAllData(request.fromUser);
+
+            if (data) {
+                result.push({
+                    ...request,
+                    fromUserData: data,
+                });
+            }
+        }
+        setFriendRequests(result);
+    };
+
+    // load friend requests on component mount
+    useEffect(() => {
+        loadFriendRequests();
+    }, []);
+
+    // accept a friend request and reload display
+    const handleAcceptFriend = async (fromId: number) => {
+        await acceptFriendRequest(fromId);
+
+        await loadFriendRequests();
+    };
+
+    // decline a friend request and reload display
+    const handleDeclineFriend = async (fromId: number) => {
+        await declineFriendRequest(fromId);
+
+        await loadFriendRequests();
```

Comment:
Similar nit to my [earlier comment](https://github.com/MetaU-25-Find-a-Friend/Find-a-Friend/pull/20/files#r2178620869).